test/integ/quote-gouda.test.ts

```diff
@@ -254,6 +256,24 @@ describe('quoteUniswapX', function () {
 
   for (const type of ['EXACT_INPUT', 'EXACT_OUTPUT']) {
     describe(`${ID_TO_NETWORK_NAME(1)} ${type} 2xx`, () => {
+      const greenlistCarveout: [Currency, Currency][] = new Array<[Currency, Currency]>();
+      for (let i = 0; i < GREENLIST_USD_STABLES.length; i += 2) {
+        const [tokenIn, tokenOut] = GREENLIST_USD_STABLES.slice(i, i + 2);
+        greenlistCarveout.push([tokenIn, tokenOut]);
+      }
+      for (let i = 0; i < GREENLIST_EUR_STABLES.length; i += 2) {
+        const [tokenIn, tokenOut] = GREENLIST_EUR_STABLES.slice(i, i + 2);
+        greenlistCarveout.push([tokenIn, tokenOut]);
+      }
+
+      const greenlistTokensPairs: [Currency, Currency][] = new Array<[Currency, Currency]>();
+      for (let i = 0; i < GREENLIST_TOKENS.length; i += 2) {
+        const [tokenIn, tokenOut] = GREENLIST_TOKENS.slice(i, i + 2);
+        if (!greenlistCarveout.find((pair) => pair[0].equals(tokenIn) && pair[1].equals(tokenOut))) {
+          greenlistTokensPairs.push([tokenIn, tokenOut]);
+        }
+      }
```

Comment:
since this is repeated lets move it into a helpers function like `getPortionGreenlistTokenPairs`